corehq/motech/repeaters/static/repeaters/js/bootstrap5/repeat_record_report.js

```diff
@@ -157,49 +157,47 @@ hqDefine('repeaters/js/bootstrap5/repeat_record_report', function () {
         });
 
         function performAction(action) {
-            if (isAnythingChecked()) {
+            var bulkSelection = bulkSelectionChecked();
+            if (bulkSelection || areRecordsChecked()) {
+                if (bulkSelection) { setFlag(bulkSelection); }
+                // only applies to checked items, not bulk selections
+                // leaving as is to preserve behavior
                 if (isActionPossibleForCheckedItems(action)) {
-                    $('#warning').addClass('hide');
-                    $('#not-allowed').addClass('hide');
+                    hideAllWarnings();
                     $popUp.modal('show');  /* todo B5: plugin:modal */
                 } else {
-                    $('#warning').addClass('hide');
-                    $('#not-allowed').removeClass('hide');
+                    showWarning('not-allowed');
                 }
             } else {
-                $('#warning').removeClass('hide');
-                $('#not-allowed').addClass('hide');
+                showWarning('no-selection');
             }
         }
 
-        function isAnythingChecked() {
-            var items = document.getElementsByName('xform_ids');
-
+        function bulkSelectionChecked() {
```

Comment:
This function's logic makes me think that you can't perform an action on more than one of these three sets (all, pending, cancelled). If that's true, if this reflected in the UI?

Reply:
Yes, that's true. It's handled by `repeat_record_report_selects.js`, which (along with a couple other things) allows only one of the bulk selection checkboxes to be checked at a time.